pkg/cluster/kubernetes/namespacer.go

```diff
@@ -23,21 +23,24 @@ type namespaceViaDiscovery struct {
 }
 
 // NewNamespacer creates an implementation of Namespacer
-func NewNamespacer(d discovery.DiscoveryInterface) (*namespaceViaDiscovery, error) {
-	fallback, err := getDefaultNamespace()
+func NewNamespacer(d discovery.DiscoveryInterface, defaultNamespaceOverride string) (*namespaceViaDiscovery, error) {
+	if defaultNamespaceOverride != "" {
+		return &namespaceViaDiscovery{fallbackNamespace: defaultNamespaceOverride, disco: d}, nil
+	}
+	fallback, err := getKubeconfigDefaultNamespace()
 	if err != nil {
 		return nil, err
 	}
 	return &namespaceViaDiscovery{fallbackNamespace: fallback, disco: d}, nil
 }
```

Comment:
Nitpick (sorry, I should had thought about this before). Can you make `getKubeconfigDefaultNamespace()` return `""` if a default namespace isn't found? (i.e.` ""` instead of `defaultFallbackNamespace`). A function called `getKubeconfigDefaultNamespace()` shouldn't do anything else.

Then, in `NewNamespacer()`, you can use `defaultFallbackNamespace` if nothing else worked.

```golang
func NewNamespacer(d discovery.DiscoveryInterface, defaultNamespaceOverride string) (*namespaceViaDiscovery, error) {
	if defaultNamespaceOverride != "" {
		return &namespaceViaDiscovery{fallbackNamespace: defaultNamespaceOverride, disco: d}, nil
	}
	kubeconfigDefaultNamespace, err := getKubeconfigDefaultNamespace()
	if err != nil {
		return nil, err
	}
	if kubeconfigDefaultNamespace != "" {
		return &namespaceViaDiscovery{fallbackNamespace: kubeconfigDefaultNamespace, disco: d}, nil
	}	
	return &namespaceViaDiscovery{fallbackNamespace: defaultFallBackNamespace, disco: d}, nil
}
```